lxd/cluster/config/config.go

```diff
@@ -500,6 +505,34 @@ var ConfigSchema = config.Schema{
 	//  shortdesc: Whether to automatically trust clients signed by the CA
 	"core.trust_ca_certificates": {Type: config.Bool, Default: "false"},
 
+	// lxdmeta:generate(entities=server; group=core; key=core.auth_secret_expiry)
+	// The secret is used for various cryptographic purposes, such as cookie encryption.
+	// When a given secret is older than the configured expiry, a new secret is generated.
+	//
+	// This configuration option accepts multiple space-separated values of the form `[0-9]+(S|M|H|d|w|m|y)`,
+	// where `S` is seconds, `M` is minutes, `H` is hours, `d` is days, `w` is weeks, `m` is months, and `y` is years.
+	// For example, `1d 3H` is 1 day and 3 hours.
+	//
+	// The default value is `1m` (1 month).
+	// The minimum value is `1d` (1 day).
+	// ---
+	//  type: string
+	//  scope: global
+	//  defaultdesc: `1m`
+	//  shortdesc: How long to use a given cluster secret.
+	"core.auth_secret_expiry": {Default: "1m", Validator: func(s string) error {
+		exp, err := shared.GetExpiry(time.Unix(0, 0), s)
```

Comment:
Due to the issues around date differences you mentioned the other day, would it makes things easier to have `shared.GetExpiry` also return a time.Duration (which it is already calculating internally anyway) so that you can simply test the duration of the expiry?